api/src/test/java/org/openmrs/module/openconceptlab/client/OclClientTest.java

```diff
@@ -132,4 +140,42 @@ public void parseDateFromPath_shouldParseDate() throws Exception {
 		
 		assertThat(date, is(calendar.getTime()));
 	}
+
+	@Test
+	public void getOclReleaseVersion_shouldGetOclReleaseVersion() throws IOException {
+		String subscriptionToken = "53fc72f0498a707a26e4d903c0f24c2db24d1e35";
+		String version = oclClient.getOclReleaseVersion(URL_WITH_VERSION, subscriptionToken);
+		assertThat(version, is("v1.0"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorrectExportUrlWhenVersionIsPassedToUrl() throws URIException {
+		String collectionVersion = "v1.0";
+		GetMethod getMethodUrlWithVersion = oclClient.getExportUrl(URL_WITH_VERSION, collectionVersion);
+		assertThat(getMethodUrlWithVersion.getURI().toString(), is(URL_WITH_VERSION + "/export"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorrectExportUrlWhenVersionIsNotPassedToUrl() throws URIException {
+		String collectionVersion = "v1.0";
+		GetMethod getMethodUrlWithoutVersion = oclClient.getExportUrl(URL_WITHOUT_VERSION, collectionVersion);
+		assertThat(getMethodUrlWithoutVersion.getURI().toString(), is(URL_WITHOUT_VERSION + "/v1.0" + "/export"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorretExportUrlWhenVersionIsNotPassedToUrl() throws URIException {
+		String collectionVersion = "v2.0";
+		GetMethod getMethod = oclClient.getExportUrl(URL, collectionVersion);
+		assertThat(getMethod.getURI().toString(), is(URL + "v2.0/export"));
+	}
+
+	@Test
+	public void removeLastUrlForwardSlashIfExist_shouldRemoveLastForwardSlashIfItExist() {
+		String newUrl = oclClient.removeLastUrlForwardSlashIfExist(URL);
+		int numberOfForwardSlashesInUrl = StringUtils.countMatches(URL, "/");
+
+		int numberOfForwardSlashesInNewUrl = StringUtils.countMatches(newUrl, "/");
+
+		assertThat(numberOfForwardSlashesInUrl, greaterThan(numberOfForwardSlashesInNewUrl));

```

Comment:
If a bug is introduced in removeLastUrlForwardSlashIfExist and it removes all slashes, will this test catch it? 😊

Reply:
I had the assertion to ensure the number of slashes is not null in my previous commit which you commented out that I should remove it

Reply:
Wouldn't what am asking be achieved by simply asserting that the difference in the number of slashes is one? 😊

Reply:
What is the value of these two assertions?

Reply:
I already added this **"Wouldn't what am asking be achieved by simply asserting that the difference in the number of slashes is one?"**

Reply:
Let me ask it differently, is there any possibility of the second assertion passing and the one before it failing?

Reply:
That can't happen

Also I think we've done lot's of discussions here and I think if the implementation doesn't seem right I could move this issue to talk and let people provide their opinions on how best this feature can be implemented instead of the back and forth.

Reply:
Because it can't happen, as you have said, that is exactly why am saying that the first assertion is redundant. Hence it should be removed. Do you agree? 😊